languages/csharp/exercises/concept/properties/.meta/design.md

```diff
@@ -0,0 +1,193 @@
+# Design
+
+## Goal
+
+The goal of this exercise is to introduce the student to C# properties and to teach the student how the concept of properties is implemented in [C#][docs.microsoft.com-properties]. We'll teach the student about properties by having the student work with classes that expose contrasting forms of properties. The students will learn to work with properties with backing fields and auto-implemented properties.
+
+Properties are covered early in the C# track as their purpose and power can be shown with few dependencies (classes, access modifiers and fields of simple types).
+
+
+## Things to teach
+
+- public properties
+- properties with backing fields
+- auto-implemented properties 
+- input verification
+- calculated properties
+- get-only properties vs. properties with private set accessors
+- set-only properties vs. properties with private get accessors
+- properties exposing reference types
+
+There is no way to enforce the use of backing fields in preference to auto-implemented properties
```

Comment:
This is correct. We _do_ have a way of detecting this more complex syntax stuff, and that is through the [C# analyzer](https://github.com/exercism/csharp-analyzer). The analyzer will be able to detect these kinds of things, and will probably be vital to this exercise for this exact reason. You can add the things the analyzer should check to the `## Analyzer` section.

Reply:
It was more about scenarios that pointed the student in a particular direction.  See https://github.com/exercism/v3/pull/973#discussion_r397092671

Reply:
Oh right. Yeah that is very hard in this case, as the student _can_ solve it differently and we can't enforce that. The only thing we can do is to make the instructions as clear as possible.

Reply:
do you think we can go as far as suggesting that they use an auto-implemented property where there is an open choice?  I would not do that if we were interested in craftsmanship as it eliminates a learning opportunity but here we just want to make sure they are familiar with all the tools in the toolkit.

Reply:
I'm not sure we should go that far, as indeed it will leave very little option to the student to learn things. I could imagine us suggesting to use an auto-implemented property in the `hints.md` file.

Reply:
I think it will be ok.  Some of the syntax will have to be absorbed by inference (i.e. we push an example of a private get but not a private set and some of the story telling is a bit strained but needs must.